x-pack/legacy/plugins/lens/public/indexpattern_plugin/state_helpers.test.ts

```diff
@@ -437,4 +445,174 @@ describe('state_helpers', () => {
+      const layer: IndexPatternLayer = {
+        columnOrder: ['col1', 'col2'],
+        columns: {
+          col1: {
+            dataType: 'string',
+            isBucketed: true,
+            label: '',
+            operationType: 'date_histogram',
+            sourceField: 'fieldC',
+            params: {
+              interval: 'd',
+            },
+          },
+          col2: {
+            dataType: 'number',
+            isBucketed: false,
+            label: '',
+            operationType: 'avg',
+            sourceField: 'fieldB',
+          },
+        },
+        indexPatternId: 'original',
+      };
+      const updatedLayer = updateLayerIndexPattern(layer, indexPattern);
+      expect(updatedLayer.columnOrder).toEqual(['col2']);
+      expect(updatedLayer.columns).toEqual({
+        col2: layer.columns.col2,
+      });
+    });
+
+    it('should remove operations referencing fields with wrong field types', () => {
+      const layer: IndexPatternLayer = {
+        columnOrder: ['col1', 'col2'],
+        columns: {
+          col1: {
+            dataType: 'string',
+            isBucketed: true,
+            label: '',
+            operationType: 'terms',
+            sourceField: 'fieldA',
+            params: {
+              orderBy: { type: 'alphabetical' },
+              orderDirection: 'asc',
+              size: 3,
+            },
+          },
+          col2: {
+            dataType: 'number',
+            isBucketed: false,
+            label: '',
+            operationType: 'avg',
+            sourceField: 'fieldC',
+          },
+        },
+        indexPatternId: 'original',
+      };
+      const updatedLayer = updateLayerIndexPattern(layer, indexPattern);
+      expect(updatedLayer.columnOrder).toEqual(['col1']);
+      expect(updatedLayer.columns).toEqual({
+        col1: layer.columns.col1,
+      });
+    });
+
+    it('should remove operations referencing fields with incompatible restrictions', () => {
+      const layer: IndexPatternLayer = {
+        columnOrder: ['col1', 'col2'],
+        columns: {
+          col1: {
+            dataType: 'string',
+            isBucketed: true,
+            label: '',
+            operationType: 'terms',
+            sourceField: 'fieldA',
+            params: {
+              orderBy: { type: 'alphabetical' },
+              orderDirection: 'asc',
+              size: 3,
+            },
+          },
+          col2: {
+            dataType: 'number',
+            isBucketed: false,
+            label: '',
+            operationType: 'min',
+            sourceField: 'fieldC',
+          },
+        },
+        indexPatternId: 'original',
+      };
+      const updatedLayer = updateLayerIndexPattern(layer, indexPattern);
+      expect(updatedLayer.columnOrder).toEqual(['col1']);
+      expect(updatedLayer.columns).toEqual({
+        col1: layer.columns.col1,
+      });
+    });
```

Comment:
I think I commented on a couple cases that I'd like to see tests for, but to reiterate:

* Name matches, but different type
* Regular -> rollup should add params
* Rollup -> regular should remove params
* fixed and calendar intervals